src/odemis/dataio/tiff.py

```diff
@@ -2613,3 +2624,121 @@ def _iterDirectories(tiff_file):
             tiff_file.ReadDirectory()
             dir_index += 1
             yield dir_index
+
+
+ 
+### FEI Metadata parsing
+def _parseINIMetadataAsDict(metadata: str, delimeter: str = "\r\n"):
+    """
+    Parses ini metadata as dictionary
+    metadata (str): metadata string as ini
+    delimeter (str): delimeter between rows (optional, default: "\r\n")
+    return (dict): metadata as dictionary
+    """
+    md = {}
+    for item in metadata.split(delimeter):
+        if item == "":
+            continue # skip blank lines
+        elif re.match(r"\[(.*?)\]", item):
+            subsystem =  item[1:-1] # rm brackets
+            md[subsystem] = {}
+        else:
+            k, v = item.split("=")   
+            md[subsystem][k] = v
+    return md
+
+
+def _convertFEItoOdemisMD(metadata):
+    """
+    Converts FEI metadata to odemis metadata format
+    metadata (str): metadata string as ini (FEI format)
+    return (dict): metadata as dictionary (odemis format)
+    """
+
+    # parse metadata as dictionary
+    fei_md = _parseINIMetadataAsDict(metadata)
+
+    # convert to odemis format
+    md = {}
+    
+    # general
+    datetime_str = fei_md["User"]["Date"] + " " + fei_md["User"]["Time"]
+    date = datetime.strptime(datetime_str, "%m/%d/%Y %I:%M:%S %p")
+    md[model.MD_ACQ_DATE] = date.timestamp()
+    
+    md[model.MD_HW_NAME] = fei_md["System"]["SystemType"]+"-"+fei_md["System"]["Dnumber"]    
+    md[model.MD_SW_VERSION] = fei_md["System"]["Software"]
+    md[model.MD_PIXEL_SIZE] = (float(fei_md["Scan"]["PixelWidth"]), 
+                                  float(fei_md["Scan"]["PixelHeight"]))
+    
+    md[model.MD_DWELL_TIME] = float(fei_md["Scan"]["Dwelltime"])
+    md[model.MD_DIMS] = "YX"
+    md[model.MD_POS] = (-float(fei_md["Stage"]["StageX"]), 
+                           -float(fei_md["Stage"]["StageY"]),
+                        #    -float(fei_md["Stage"]["StageZ"])
+                           )
+
+    # beam metadata
+    beam_type = fei_md["Beam"]["Beam"]
+    scan_type = fei_md["Beam"]["Scan"]
+
+    md[model.MD_EBEAM_VOLTAGE] = float(fei_md[beam_type]["HV"])
+    md[model.MD_EBEAM_CURRENT] = float(fei_md[beam_type]["BeamCurrent"])
+    # NB: this is not actual spot size, actuall spot diam is in xml metadata
+    md[model.MD_EBEAM_SPOT_DIAM] = float(fei_md["Beam"]["Spot"]) 
+    
+    # acquistion type
+    acq_map =  {
+        "EBeam": model.MD_AT_EM,
+        "IBeam": model.MD_AT_FIB,
+    }
+    md[model.MD_ACQ_TYPE] = acq_map[beam_type]
+
+    # detector
+    md[model.MD_DET_TYPE] = (model.MD_DT_NORMAL 
+                                if fei_md["Detectors"]["Name"] in ["ETD", "TLD"] 
+                                else model.MD_DT_INTEGRATING)
+    
+    # system components
+    extra_md = {}
```

Comment:
Note that this EXTRA_SETTINGS metadata is normally only used as info for the user (containing detailed settings about the hardware properties during an acquisition). It's never used for an actual functionality in Odemis). So it would be fine to completely skip it. Alternatively, you could also just copy-paste the whole "INI" string into it and be done. 
I don't think that trying to convert it into something that looks like it was acquired by Odemis is useful.

Reply:
Ok, removing it for clarity